### ISSUE TITLE
Remote Git Remote Repository

### ISSUE THREAD

**Issue by jheare** [sr320/fish546-2015#5]
How do you create a remote repository on a server such as Hummingbird to access when I'm running material on the cloud server?

**Comment by sr320** [comment]
Prob easiest to get it to work via terminal sitting at hummingbird -- remote execution would be simple.

**Comment by jheare** [comment]
Actually I was able to do all of this remotely but I'm still working out some kinks. I'll post the workthrough for remote set up when I make the robust solution that doesn't incur useless redundancies.

**Comment by jheare** [comment]
So the easiest way I've found to set up a remote repository on a server is this:
1. Make a new or use an existing repository on Github. For my efforts I'm using the Fish546-Jake repository.
2. Clone the repository using the command:
   **git clone [URL of Github Repository]**
   You can find this url on the lower right hand corner of the github repository page above the download link. I suggest using the HTTPS option as its the one that I've gotten to work. This will make an exact duplicate of everything in the repository. Do not place brackets around the url. 
3. Check your repository remote name with:
   **git remote**
   This should show that the repository remote alias is "origin". This name can be changed but for our purposes that is unnecessary. Alternate aliases can be created as well.
4. Check your repository status with:
   **git status**
   This will tell you if you have any untracked files or changes that need to be added or committed. At this point there should be nothing that needs adding or committing, so we'll make a new ipynb file to commit to the repository. 
5. Launch ipython notebook from your directory, create a new notebook, save notebook, and shutdown ipython. I'll write up a separate entry for how to operate ipython notebook from a remote server as its not hard but unintuitive.
6. Check your repository status again with:
   **git status**
   You should see that there is a folder for ipynb_checkpoints and a file for your new notebook listed as  untracked changes. 
7. Add the untracked files to the repository using:
   **git add -A**
   This will add all untracked files. If you have files or folders you don't want to add, you can add each file/folder individually but for our purposes add all is the best option. 
8. Commit the newly added files to the repository with a message explaining the commit with:
   **git commit -a -m 'your message here'**
   The -a commits all the added files, again you can specify them individually but it best to do them all. The -m tells the commit function that the following statement in quotation marks is the commit message. If you fail to do this, git commit will automatically open Vi/Vim. Vi/Vim is a terminal text editor and comes with its own set of operators and functions. If you accidentally open Vi/Vim don't worry, just use your directional buttons to move the cursor to the last available line (it will beep at you when you try to go beyond that line) from there you can delete the hash mark and type your description. Once you finish typing the description hit ESC once followed by SHIFT+ZZ (on Windows) to save your comment and quit Vi/Vim. Once this is done the files will be committed to the repository.
9. Push the commits to github with:
   **git push origin master**
   Once you do this github will ask for your github username and password. Origin designates your servers local repository alias while master designates the master portion of the repository on github. If you want you can replace master with any name and github will generate a new branch on the repository. Now you should have a copy of all your servers project files on the appropriate github repository. 
10. After adding new notebooks, updating current notebooks, or producing any new files in the server repository **YOU MUST RE RUN STEPS 7,8,9** to add, commit, and push the files from the server to github. 

Also if something changes on the github master, the push command may fail because the server's repository is not up-to-date. You can quickly fix this with the command:
**git fetch origin; git merge origin/master**
Where origin is the name of the servers working repository. This will download any new changes to the repository and clear the way for you to push any local commits you have. You can then just rerun the push command from step 9 and this will update githubs repository. 

You can find a more indepth guide to using remote functions at [gitref.org](http://gitref.org/remotes/#remote).
